sdk/rust-language-support/libveracruz/src/host.rs

```diff
@@ -354,6 +354,16 @@ extern "C" {
     /// Fills a buffer with random bytes taken from a platform-specific trusted
     /// entropy source.
     fn __veracruz_hcall_getrandom(buffer: *mut u8, size: u32) -> i32;
+    /// Reads previous result the given buffer
+    fn __veracruz_hcall_read_previous_result(buffer: *mut u8, size: u32) -> i32;
+    /// Reads the size of previous result encoded by pinecone
```

Comment:
These should not be encoded by `pinecone`?  The use of pinecone should not be baked into the Veracruz ABI, as it's just an incidental implementation choice in some of our examples.  Rather, data should be passed across the ABI boundary as little endian encoded byte patterns, similar to how the existing ABI functions are implemented.  Really, I don't think Mexico City or `libveracruz` should have *any* dependence on the `pinecone` library at all.

Reply:
I meant to use an option type to distinguish if there is a previous result. Here I use pinecone to flatten down to bytes. I think a better way is to pass a boolean for that.

Reply:
I think a boolean or other error code to indicate a parameter is defined, or not, is a good idea.  All ABI functions should be easily usable from other languages other than Rust (e.g. C) in a straightforward way.  Adopting C-style conventions is therefore a good idea.